pkg/specgen/generate/pod_create.go

```diff
@@ -2,22 +2,107 @@ package generate
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"net"
+	"os/exec"
+	"path/filepath"
 
 	"github.com/containers/common/pkg/config"
 	"github.com/containers/podman/v3/libpod"
 	"github.com/containers/podman/v3/libpod/define"
 	"github.com/containers/podman/v3/pkg/domain/entities"
 	"github.com/containers/podman/v3/pkg/rootless"
 	"github.com/containers/podman/v3/pkg/specgen"
+	"github.com/containers/podman/v3/pkg/specgenutil"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+func makeInfraSpec(p *entities.PodSpec, rt *libpod.Runtime) error {
+	if p.PodSpecGen.NoInfra {
+		return nil
+	}
+
+	rtConfig, err := rt.GetConfigNoCopy()
+	if err != nil {
+		return err
+	}
+
+	// NOTE: we need pull down the infra image if it was explicitly set by
+	// the user (or containers.conf) to the non-default one.  We also need
+	// to pull the image down if we're running in rootless mode without
+	// fuse-overlayfs; a requirement to run an overlay rootfs.
+	mountPause := false
+	imageName := p.PodSpecGen.InfraImage
+	if imageName == "" {
+		imageName = rtConfig.Engine.InfraImage
+	}
+
+	pullImage := imageName != config.DefaultInfraImage
+	if !pullImage && rootless.IsRootless() {
+		if _, err := exec.LookPath("fuse-overlayfs"); err != nil {
+			pullImage = true
+		}
+	}
+
+	if pullImage {
+		_, err := rt.LibimageRuntime().Pull(context.Background(), imageName, config.PullPolicyMissing, nil)
+		if err != nil {
+			return err
+		}
+	} else {
+		pausePath, err := rtConfig.FindHelperBinary("pause/pause", false)
+		if err != nil {
+			return fmt.Errorf("finding pause binary: %w", err)
+		}
+		imageName = filepath.Dir(pausePath) + ":O"
+		mountPause = true
+	}
+
+	p.PodSpecGen.InfraContainerSpec = specgen.NewSpecGenerator(imageName, mountPause)
+	infraOptions := &entities.ContainerCreateOptions{
```

Comment:
hold on, what happened to the infraOptions in cmd/podman/pods/create.go? Mapping those over when they are given is the best way to do this. It allows us to basically drop container create options into pod create with minimal change. Does this undo the `FillOutSpecGen` in cmd/podman?